### ISSUE TITLE
type mismatch error -- add checks for `by()` 

### ISSUE THREAD

**Issue by mkaulisch** [asjadnaqvi/stata-bumparea#2]
I do get an -type mismatch- error with version 1.11. And I do not know why:
`[CODE]
* Example generated by -dataex-. For more info, type help dataex
clear
input byte(field period) double impact_p_r
3  1 3432
3  2 3569
3  3 3707
3  4 3997
3  5 4276
3  6 4534
3  7 4830
3  8 5082
3  9 5410
3 10 5797
3 11 6034
3 12 6248
3 13 6864
4  1  357
4  2  394
4  3  405
4  4  427
4  5  459
4  6  504
4  7  548
4  8  593
4  9  635
4 10  641
4 11  675
4 12  689
4 13  686
5  1  152
5  2  144
5  3  158
5  4  161
5  5  146
5  6  159
5  7  154
5  8  162
5  9  173
5 10  174
5 11  179
5 12  189
5 13  217
6  1  907
6  2  928
6  3  947
6  4  962
6  5  995
6  6 1037
6  7 1047
6  8 1112
6  9 1115
6 10 1160
6 11 1224
6 12 1180
6 13 1191
7  1  137
7  2  168
7  3  198
7  4  227
7  5  261
7  6  320
7  7  356
7  8  399
7  9  445
7 10  456
7 11  475
7 12  523
7 13  578
end
label values field l_field_kurz
label def l_field_kurz 3 "Biomedical & health sciences", modify
label def l_field_kurz 4 "Life & earth sciences", modify
label def l_field_kurz 5 "Mathematics & computer science", modify
label def l_field_kurz 6 "Physical sciences & engineering", modify
label def l_field_kurz 7 "Social sciences & humanities", modify
label values period l_period
label def l_period 1 "2006–2009", modify
label def l_period 2 "2007–2010", modify
label def l_period 3 "2008–2011", modify
label def l_period 4 "2009–2012", modify
label def l_period 5 "2010–2013", modify
label def l_period 6 "2011–2014", modify
label def l_period 7 "2012–2015", modify
label def l_period 8 "2013–2016", modify
label def l_period 9 "2014–2017", modify
label def l_period 10 "2015–2018", modify
label def l_period 11 "2016–2019", modify
label def l_period 12 "2017–2020", modify
label def l_period 13 "2018–2021", modify
[/CODE]

bumparea impact_p_r period , by(field) ///
 name(bump_field_voll, replace) `

**Comment by asjadnaqvi** [comment]
The `by()` term is expecting a string input somewhere in the code. I will update it. 

For now, this works:

```
decode field, gen(field2)

bumparea impact_p_r period , by(field2) 
```

**Comment by asjadnaqvi** [comment]
fixed v1.2
